client.go

```diff
@@ -78,23 +86,29 @@ func NewClient(config *ClientConfig) (*Client, error) {
 
 		client.HTTPClient.Transport = &http.Transport{TLSClientConfig: conf}
 	}
-	return &Client{
-		username: config.Username,
-		password: config.Password,
-		baseURL:  config.BaseURL,
-		client:   client,
-	}, nil
+	c := &Client{
+		username:     config.Username,
+		password:     config.Password,
+		baseURL:      config.BaseURL,
+		client:       client,
+		securityPath: defaultSecurityAPIPath,
+	}
+	if verifyConnection {
+		return c, c.setSecurityPath(ctx)
```

Comment:
Maybe it makes more sense to log a warning if the elasticsearch version can't be determined, and leave the securityPath at the default, instead of returning an error? I could see it being a bad surprise for folks that don't have the "monitor" permission set on the elasticsearch role.

Reply:
Is verifyConnection something that's only applicable with ES 7+?

Reply:
@tomhjp So I played around with this a bit, and I'm not sure we can tell if a mount config is new or existing inside a database plugin. Specifically because the Initialize() function doesn't have a view of the storage backend, just the InitializeRequest{} struct: https://github.com/hashicorp/vault-plugin-database-elasticsearch/blob/cbe7ef731353f29083c719384bb3786c33f168a1/elasticsearch.go#L72-L74

I think Initialize() is being called on the vault side here: https://github.com/hashicorp/vault/blob/67ca3beb1ecde48a5a6a8befb783b1b95aa5ad8e/builtin/logical/database/version_wrapper.go#L49-L51